src/mlagility/analysis/status.py

```diff
@@ -26,22 +26,40 @@ def recursive_print(
 ) -> None:
     script_names_visited = []
 
-    for h in models_found.keys():
-        if parent_hash == models_found[h].parent_hash and models_found[h].executed > 0:
-            print_file_name = models_found[h].script_name not in script_names_visited
+    for model_hash in models_found.keys():
+        workloads_executed = False
+        for workload_hash in models_found[model_hash].workloads.keys():
+            workload = models_found[model_hash].workloads[workload_hash]
```

Comment:
Architectural question (I'll stop reviewing the code until we bottom out on this): what are we gaining from maintaining a list of models and then workloads per model? Why not just replace the `models_found` list with a `workloads_found` list?

I can imagine that it would be helpful to organize the filesystem along model/workload lines, organize the results that way, etc. but none of that appears to be happening yet.

Reply:
The idea of organizing the results that way sounds great. What do you think about the format below?

```
two_input_shapes.py:
        model (executed 2x - 0.04s)
                Model Type:     Pytorch (torch.nn.Module)
                Class:          EfficientNet (<class 'timm.models.efficientnet.EfficientNet'>)
                Location:       /home/dhnoronha/.local/lib/python3.8/site-packages/timm/models/efficientnet_builder.py, line 472
                Parameters:     1,677,128 (3.2 MB)

                Workload A (executed 1x - 0.01s)
                          Input Shape:    Positional Arg 1': (1, 3, 28, 28)
                          Hash:           b63f3042
                          Status:         Successfully benchmarked on Intel(R) Xeon(R) CPU @ 2.20GHz (ort v1.14.1)
                                          Mean Latency:   5.540   milliseconds (ms)
                                          Throughput:     180.5   inferences per second (IPS)

                Workload B (executed 1x - 0.03s)
                          Input Shape:    'Positional Arg 1': (1, 3, 224, 224)
                          Hash:           062ae984
                          Status:         Successfully benchmarked on Intel(R) Xeon(R) CPU @ 2.20GHz (ort v1.14.1)
                                          Mean Latency:   5.540   milliseconds (ms)
                                          Throughput:     180.5   inferences per second (IPS)
```

instead of 

```
two_input_shapes.py:
        model (executed 1x)
                Model Type:     Pytorch (torch.nn.Module)
                Class:          EfficientNet (<class 'timm.models.efficientnet.EfficientNet'>)
                Location:       /home/dhnoronha/.local/lib/python3.8/site-packages/timm/models/efficientnet_builder.py, line 472
                Parameters:     1,677,128 (3.2 MB)
                Input Shape:    {'Positional Arg 1': (1, 3, 28, 28)}
                Hash:           b63f3042
                Status:         Successfully benchmarked on Intel(R) Xeon(R) CPU @ 2.20GHz (ort v1.14.1)
                                Mean Latency:   5.540   milliseconds (ms)
                                Throughput:     180.5   inferences per second (IPS)

        model (executed 1x)
                Model Type:     Pytorch (torch.nn.Module)
                Class:          EfficientNet (<class 'timm.models.efficientnet.EfficientNet'>)
                Location:       /home/dhnoronha/.local/lib/python3.8/site-packages/timm/models/efficientnet_builder.py, line 472
                Parameters:     1,677,128 (3.2 MB)
                Input Shape:    {'Positional Arg 1': (1, 3, 224, 224)}
                Hash:           062ae984
                Status:         Successfully benchmarked on Intel(R) Xeon(R) CPU @ 2.20GHz (ort v1.14.1)
                                Mean Latency:   24.669  milliseconds (ms)
                                Throughput:     40.5    inferences per second (IPS)

```

Reply:
This looks really nice and clean. Only question is how the workloads are going to be named? Is there a nice, non-confusing way to name them?

Further question: how difficult is it going to be to access these workloads in the cache? Am I going to have to copy paste together `instancename_workloadname_hash`?